test/bin/setup-e2e.sh

```diff
@@ -46,11 +53,32 @@ for i in $(seq 1 "${number}"); do
   fi
 
   cp -r $project_dir/hack/deploy/controlplane/* $deploy_dir
+
+  # expose apiserver
   sed -i "s/API_HOST/${external_host_ip}/" $deploy_dir/deployment.yaml
   sed -i 's/ClusterIP/NodePort/' $deploy_dir/service.yaml
   sed -i '/route\.yaml/d' $deploy_dir/kustomization.yaml
   sed -i "/targetPort.*/a  \ \ \ \ \ \ nodePort: $external_host_port" $deploy_dir/service.yaml
 
+  # setup external etcd 
+  sed -i '/pvc\.yaml/d' $deploy_dir/kustomization.yaml
+  sed -i '/disableNameSuffixHash.*/a patches:' $deploy_dir/kustomization.yaml
+  sed -i '/patches.*/a \ \ - external-etcd-patch.yaml' $deploy_dir/kustomization.yaml
+  sed -i '/^secretGenerator.*/a \ \ - cert-etcd/client-key.pem' $deploy_dir/kustomization.yaml
+  sed -i '/^secretGenerator.*/a \ \ - cert-etcd/client.pem' $deploy_dir/kustomization.yaml
+  sed -i '/^secretGenerator.*/a \ \ - cert-etcd/ca.pem' $deploy_dir/kustomization.yaml
+  sed -i '/^secretGenerator.*/a \ \ files:' $deploy_dir/kustomization.yaml
+  sed -i '/^secretGenerator.*/a - name: cert-etcd' $deploy_dir/kustomization.yaml
+  etcd_size=$(${KUBECTL} -n ${etcdns} get statefulset.apps/etcd -o jsonpath='{.spec.replicas}')
+  etcd_services=http://etcd-0.etcd.${etcdns}:2379
+  for((i=1;i<$etcd_size;i++))
+  do
+    etcd_services=${etcd_services}",http://etcd-"$i".etcd.${etcdns}:2379"
+  done
+  sed -i "s@http://127.0.0.1:2379@${etcd_services}@g" $deploy_dir/external-etcd-patch.yaml
+  sed -i "s/--etcd-prefix=.*/--etcd-prefix=${namespace}\"/g" $deploy_dir/external-etcd-patch.yaml
+  sed -i "s/API_HOST/${external_host_ip}/g" $deploy_dir/external-etcd-patch.yaml
+        
```

Comment:
Can we setup a separated case to test external etcd? I mean we test 2 cases: external etcd and embedded etcd.

Reply:
For now, we use the external etcd in the e2e test and use the embedded etcd with the integration test.

So there is a scenario for external etcd in the tests.

Reply:
sound good. thanks.